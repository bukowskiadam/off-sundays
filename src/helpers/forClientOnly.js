import React from 'react';

import getDisplayName from './getDisplayName';

export default (Component, Placeholder = null) => {
    class ForClientOnly extends React.Component {
        state = {
            client: false,
        };

        componentDidMount() {
            this.setState({
                client: true,
            });
        }

        renderPlaceholder = () => (Placeholder ? <Placeholder /> : null);

        render() {
            return this.state.client ? <Component {...this.props} /> : this.renderPlaceholder();
        }
    }

    const ComponentName = getDisplayName(Component);
    const PlaceholderName = Placeholder ? getDisplayName(Placeholder) : 'null';
    ForClientOnly.displayName = `ForClientOnly(${ComponentName}, ${PlaceholderName}`;

    return ForClientOnly;
};
