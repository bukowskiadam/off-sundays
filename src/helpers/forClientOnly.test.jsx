import React from 'react';
import { mount, render } from 'enzyme';

import forClientOnly from './forClientOnly';

describe('forClientOnly', () => {
    const ClientComponent = () => <div>Client Component</div>;
    const Placeholder = () => <div>Placeholder</div>;

    it('renders placeholder correctly for server side', () => {
        const Component = forClientOnly(ClientComponent, Placeholder);

        const wrapper = render(<Component />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders nothing correctly for server side', () => {
        const Component = forClientOnly(ClientComponent);

        const wrapper = render(<Component />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly for client side', () => {
        const Component = forClientOnly(ClientComponent, Placeholder);

        const wrapper = mount(<Component />);

        expect(wrapper).toMatchSnapshot();
    });
});
