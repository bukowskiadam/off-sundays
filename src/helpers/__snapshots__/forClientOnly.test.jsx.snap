// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forClientOnly renders correctly for client side 1`] = `
<ForClientOnly(ClientComponent, Placeholder>
  <ClientComponent>
    <div>
      Client Component
    </div>
  </ClientComponent>
</ForClientOnly(ClientComponent, Placeholder>
`;

exports[`forClientOnly renders nothing correctly for server side 1`] = `null`;

exports[`forClientOnly renders placeholder correctly for server side 1`] = `
<div>
  Placeholder
</div>
`;
